nodejs/action.yaml

```diff
@@ -26,12 +26,36 @@ runs:
         node-version-file: '.nvmrc'
       env:
         NPM_TOKEN: ${{ inputs.NPM_REGISTRY_TOKEN }}
-    - name: npm install, build, and test
+    - name: npm install
       run: |
         npm ci
+      shell: bash
+      env:
+        CI: "true"
+        NPM_TOKEN: ${{ inputs.NPM_REGISTRY_TOKEN }}
+    - name: npm audit
+      run: |
         npm run vulnerabilities
+      shell: bash
+      env:
+        CI: "true"
+        NPM_TOKEN: ${{ inputs.NPM_REGISTRY_TOKEN }}
+    - name: npm lint
+      run: |
         npm run lint:report
+      shell: bash
+      env:
+        CI: "true"
+        NPM_TOKEN: ${{ inputs.NPM_REGISTRY_TOKEN }}
+    - name: npm copy paste detection
+      run: |
         npm run cpd
+      shell: bash
+      env:
+        CI: "true"
+        NPM_TOKEN: ${{ inputs.NPM_REGISTRY_TOKEN }}
+    - name: npm test
+      run: |        
```

Comment:
```yaml
strategy:
      matrix:
        action: ["run vulnerabilities", "lint:report", "test"]
        // ...
steps:
       - name: npm actions
       run: |
          npm ${{matrix.action}}
       env:
            CI: "true"
            NPM_TOKEN: ${{inputs.NPM_REGISTRY_TOKEN}}
```

Reply:
you do loose the ability to check on error but we can split the action into an object and do some magic? continue-on-error doesnt work in `composite` actions (maybe now?)

```yaml
strategy:
      matrix:
        action: [{"op":"run vulnerabilities", "error": true}, {"op":"lint:report", "error": false}, {"op":"test", "error":false}]
        // ...
steps:
       - name: npm actions
       run: |
          npm ${{matrix.action.op}}
       env:
            CI: "true"
            NPM_TOKEN: ${{inputs.NPM_REGISTRY_TOKEN}}
         continue-on-error: ${{matrix.action.error}}
```